src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -39,12 +54,36 @@ export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
   scheduleStartHour,
   personalizedEvent: isPersonalizedEvent = false,
 }) => {
+  const eventRef: RefObject<HTMLDivElement> = useRef(null);
+  const isMinimallyLongEvent = event.duration_minutes >= MIN_LONG_EVENT_LENGTH;
+  const isShortEvent = event.duration_minutes <= SHORT_EVENT_LENGTH;
+  const isEventLong = event.duration_minutes >= LONG_EVENT_LENGTH;
+
   const { userId } = useUser();
+  const {
+    isShown: isExpanded,
+    hide: hideExpanded,
+    show: showExpanded,
+  } = useShowHide();
+  const {
+    isShown: isExpandedStatic,
+    toggle: toggleExpandedStatic,
+    hide: hideExpandedStatic,
+  } = useShowHide();
+
+  const isBookmarkDisplayed =
+    isMinimallyLongEvent || isExpanded || isExpandedStatic;
+  const isContentDisplayed = isExpanded || isExpandedStatic || isEventLong;
+  const isShowExpand = (!isShortEvent || isExpanded) && !isEventLong;
 
   // @debt ONE_HOUR_IN_MINUTES is deprectated; refactor to use utils/time or date-fns functions
-  const eventWidthPx =
-    (event.duration_minutes * SCHEDULE_HOUR_COLUMN_WIDTH_PX) /
-    ONE_HOUR_IN_MINUTES;
+  const eventWidthPx = getMinuteValue(
+    event.duration_minutes * SCHEDULE_HOUR_COLUMN_WIDTH_PX
+  );
+
+  const expandedEventPx = getMinuteValue(
+    LONG_EVENT_LENGTH * SCHEDULE_HOUR_COLUMN_WIDTH_PX
+  );
```

Comment:
Why was this changed? The way it's written here is confusingly obtuse to say the least, but I think it may also just be buggy too.

Why can't we use a function from `date-fns` here instead of creating our own?

I'm having trouble wrapping my head around what the intent even is, but maybe something like:

- https://date-fns.org/v2.22.1/docs/hoursToMinutes
- https://date-fns.org/v2.22.1/docs/hoursToSeconds
- https://date-fns.org/v2.22.1/docs/minutesToHours
- etc?

Reply:
The old logic:

```typescript
(event.duration_minutes * SCHEDULE_HOUR_COLUMN_WIDTH_PX) / ONE_HOUR_IN_MINUTES
```

Removing brackets/etc:

```typescript
event.duration_minutes * SCHEDULE_HOUR_COLUMN_WIDTH_PX / ONE_HOUR_IN_MINUTES
```

Reordering to make more logical sense:

```typescript
event.duration_minutes / ONE_HOUR_IN_MINUTES * SCHEDULE_HOUR_COLUMN_WIDTH_PX 
```

Using `minutesToHours` from `date-fns`:

```typescript
minutesToHours(event.duration_minutes) * SCHEDULE_HOUR_COLUMN_WIDTH_PX 
```

Reply:
See below suggestion for applying this fix

Reply:
yep, looks good
please see https://github.com/sparkletown/sparkle/pull/1733#discussion_r665945040 so we can proceed with doing it

Reply:
fixed a265519